packages/opentelemetry-exporter/src/index.js

```diff
@@ -0,0 +1,288 @@
+   * @param {{ agentKey?: string, endpointUrl?: string }}
+   */
+  constructor({ agentKey, endpointUrl }) {
+    this._isShutdown = false;
+    // If endpoint URL and agent key are not provided, we don't set them in an attempt to use env vars, if they were set
+    if (endpointUrl) {
+      process.env[instanaEndpointUrlEnvVar] = endpointUrl;
+    }
+    if (agentKey) {
+      process.env[instanaAgentKeyEnvVar] = agentKey;
+    }
+
+    if (!process.env[instanaEndpointUrlEnvVar] || !process.env[instanaAgentKeyEnvVar]) {
+      throw new Error('The Instana endpoint URL and/or the agent key were not provided');
+    }
+
+    instanaEnvironment.validate();
+    instanaBackendConnector.init();
+  }
+
+  /**
+   * @param {Array.<ReadableSpan>} spans
+   * @param {(result: import('@opentelemetry/core').ExportResult) => void} resultCallback
+   */
+  export(spans, resultCallback) {
+    const validSpans = spans.map(this._transform.bind(this));
+
+    if (!this._isShutdown) {
+      this._sendSpans(validSpans, resultCallback);
+    }
+  }
+
+  /**
+   * @returns {Promise.<void>}
+   */
+  shutdown() {
+    /**
+     * In the future, there may be the need to "end" any kind of process or network communications with our backend.
+     * Right now, there isn't seem to be any situation where we need that, as we simply make HTTP requests without
+     * actually making any type of stateful or persistent connections.
+     */
+    this._isShutdown = true;
+    return Promise.resolve();
+  }
+
+  /**
+   * @param {Array.<InstanaBaseSpan>} spans
+   * @param {(result: import('@opentelemetry/core').ExportResult) => void} resultCallback
+   */
+  _sendSpans(spans, resultCallback) {
+    instanaBackendConnector.sendSpans(spans, (err /* , _responseBody */) => {
+      /** @type {import('@opentelemetry/core').ExportResult} */
+      let result = { code: ExportResultCode.SUCCESS };
+      if (err) {
+        result = { code: ExportResultCode.FAILED, error: err };
+      }
+      resultCallback(result);
+    });
+  }
+
+  /**
+   * @param {ReadableSpan} span
+   * @returns {InstanaBaseSpan}
+   */
+  _transform(span) {
+    /** @type {import('@instana/core/src/tracing/cls').InstanaBaseSpan} */
+    const result = {
+      n: 'otel',
+      f: {
+        e: process.pid
+      },
+      data: {}
+    };
+
+    const traceId = this._buildTraceOrSpanId(span.spanContext().traceId, true);
+
+    if (traceId.length === 32) {
+      result.t = traceId.substr(16);
+      result.lt = traceId;
+    } else {
+      result.t = traceId;
+    }
+
+    const hasParent = typeof span.parentSpanId === 'string' && span.parentSpanId.length > 0;
+
+    if (hasParent) {
+      result.p = this._buildTraceOrSpanId(span.parentSpanId);
+    }
+
+    result.s = this._buildTraceOrSpanId(span.spanContext().spanId);
+    result.ts = this._getTimeInMicroseconds(span.startTime);
+    result.d = this._getTimeInMicroseconds(span.duration);
+    let isEntrySpan = false;
+
+    switch (span.kind) {
```

Comment:
I was testing an example Otel Instrumention ([pg](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node)).

I received 4 spans (3x express + 1x pg) in the backend.
The span kind for pg query is "ENTRY". I thought this has to be EXIT. Maybe I am just confused :)

Reply:
My bad. I was sending wrong data to the backend while I was playing with the code locally. Sorry!